src/map.cpp

```diff
@@ -130,6 +130,14 @@ Map::Map(const SizedType &type) : IMap(type)
   int max_entries = 128 << 10;
   int flags = 0;
 
+  if ((value_size % 8) != 0)
+  {
+    // For maps of type BPF_MAP_TYPE_STACK_TRACE, value_size is expected to be
+    // a multiple of 8 (see stack_map_alloc in kernel/bpf/stackmap.c). Adjust
+    // if needed.
+    value_size = (value_size / 8 + 1) * 8;
+  }
+
```

Comment:
Good catch! I assumed stack trace entries would be 4 bytes on arm, but on closer inspection `struct perf_callchain_entry` contains an `u64` array (which ultimately gets `memcpy`'d into the map): [perf_event.h](https://github.com/torvalds/linux/blob/dc164f4fb00a0abebdfff132f8bc7291a28f5401/include/linux/perf_event.h#L67).

I'll update this to `uint64_t` instead of `uintptr_t`.